### PR TITLE
issues-294 Move escape and unescape string to backend

### PR DESCRIPTION
**Pull request by ikrivosheev**
## PR Info

- Closes <https://github.com/SeaQL/sea-query/issues/294>

## Adds

Add new trait: `EscapeBuilder`

## Breaking Changes

Remove `escape_string` and `unescape_string` functions.

### CODE REVIEW

File: src/value.rs [hunk 1]
```diff
@@ -1177,34 +1132,6 @@ impl Values {
...
-        assert_eq!(unescape_string(escape_string(test).as_str()), test);
-    }
-
-    #[test]
-    fn test_escape_4() {
-        let test = "a\"b";
-        assert_eq!(escape_string(test), "a\\\"b".to_owned());
-        assert_eq!(unescape_string(escape_string(test).as_str()), test);
-    }
-
```
Comment (billy1624) [inline src/value.rs:1207]: Hey @ikrivosheev, could we add back these test cases in query test?

Comment (ikrivosheev) [reply src/value.rs:1207]: @billy1624 I cannot find query test.

Comment (tyt2y3) [reply src/value.rs:1207]: Agree with Billy we'd better not delete the test cases (actually we might want more)

Comment (ikrivosheev) [reply src/value.rs:1207]: @tyt2y3 @billy1624, done, I divided the tests for each backend.

Comment (billy1624) [reply src/value.rs:1207]: Thanks! @ikrivosheev

**Review by billy1624 — APPROVED** [review]

**Pull request merged**